File: Ninja Training/Tabulation.py
def ninjaTraining(n, arr):
    dp = [ [-1 , -1 , -1] for _ in range(3)]

    for i in range(3):
        dp[0][i] = arr[0][i]
    
    for i in range(1,n):
        for j in range(3):
            maximum = 0
            for k in range(3):
                if j != k:
                    maximum = max(maximum,arr[i][j] + dp[i-1][k])
            dp[i][j] =  maximum 

    ans = 0
    for i in range(3):
        ans = max(ans,dp[n-1][i])
    return ans

if __name__ == "__main__":
    print(ninjaTraining(3,[ [18 ,11 ,19],[4, 13 ,7],[1, 8 ,13] ] ))

File: Ninja Training/Tabulation.java

public class Tabulation {
    public static void main(String[] args) {
        int[][] arr = {
            {18, 11, 19},
            {4, 13, 7},
            {1, 8, 13}
        };

        int size = arr.length;
        int[][] dp = new int[size][3];

        System.arraycopy(arr[0], 0, dp[0], 0, 3);

        for(int i = 1;i< size; i ++) {
            
            for(int j = 0; j < 3; j++)  {
                int maximum = 0;
                
                for(int k = 0 ; k < 3; k ++) {
                    if( j != k) 
                        maximum = Math.max(maximum,arr[i][j] +  dp[i-1][k]);
                }

                dp[i][j] =  maximum;
            }

        }

        int ans = 0;
        for(int i = 0; i< 3 ;i ++) {
            ans = Math.max(ans,dp[size-1][i]);
        }

        System.out.println(ans);
        
    }
}